File: app/src/main/java/com/ddu/ui/fragment/study/ui/TextViewFragment.java
package com.ddu.ui.fragment.study.ui;

import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.ddu.R;
import com.ddu.icore.ui.fragment.DefaultFragment;
import com.ddu.ui.effect.CurveFloatingPathEffect;
import com.ddu.ui.effect.CurvePathFloatingAnimator;
import com.ddu.ui.effect.ScaleFloatingAnimator;
import com.ddu.ui.view.FloatingText;
import com.ddu.ui.view.RevealTextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

/**
 * Created by yzbzz on 16/4/14.
 */
public class TextViewFragment extends DefaultFragment {


    @Nullable
    @BindView(R.id.tv_reveal_text_view)
    RevealTextView tvRevealTextView;
    @Nullable
    @BindView(R.id.tv_translateView)
    TextView tvTranslateView;
    @Nullable
    @BindView(R.id.tv_curveView)
    TextView tvCurveView;
    @Nullable
    @BindView(R.id.tv_scaleView)
    TextView tvScaleView;

    private Unbinder unbinder;

    @NonNull
    public static TextViewFragment newInstance(String taskId) {
        Bundle arguments = new Bundle();
        arguments.putString(ARGUMENT_TASK_ID, taskId);
        TextViewFragment fragment = new TextViewFragment();
        fragment.setArguments(arguments);
        return fragment;
    }

    @Override
    public void initData(Bundle savedInstanceState) {

    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_ui_textview;
    }

    @Override
    public void initView() {
        unbinder = ButterKnife.bind(this, mView);
    }

    @OnClick({R.id.tv_reveal_text_view, R.id.tv_translateView, R.id.tv_curveView, R.id.tv_scaleView})
    public void onClick(@NonNull View view) {
        switch (view.getId()) {
            case R.id.tv_reveal_text_view:
                tvRevealTextView.replayAnimation();
                break;
            case R.id.tv_translateView:
                showFloatingTextByTranslate(view);
                break;
            case R.id.tv_curveView:
                showFloatingTextByCurve(view);
                break;
            case R.id.tv_scaleView:
                showFloatingTextByScale(view);
                break;
            default:
                break;
        }
    }

    private void showFloatingTextByTranslate(View view) {
        FloatingText translateFloatingText = new FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.RED)
                .textSize(100)
                .textContent("+1000")
                .build();
        translateFloatingText.attach2Window();
        translateFloatingText.startFloating(view);
    }

    private void showFloatingTextByCurve(View view) {
        FloatingText cubicFloatingText = new FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.RED)
                .textSize(100)
                .floatingAnimatorEffect(new CurvePathFloatingAnimator())
                .floatingPathEffect(new CurveFloatingPathEffect())
                .textContent("Hello! ").build();
        cubicFloatingText.attach2Window();
        cubicFloatingText.startFloating(view);
    }

    private void showFloatingTextByScale(View view) {
        FloatingText scaleFloatingText = new FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.parseColor("#7ED321"))
                .textSize(100)
                .offsetY(-100)
                .floatingAnimatorEffect(new ScaleFloatingAnimator())
                .textContent("+188").build();
        scaleFloatingText.attach2Window();
        scaleFloatingText.startFloating(view);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/ddu/ui/fragment/study/ui/TextViewFragment.kt
package com.ddu.ui.fragment.study.ui

import android.graphics.Color
import android.os.Bundle
import android.view.View
import com.ddu.R
import com.ddu.icore.ui.fragment.DefaultFragment
import com.ddu.ui.effect.CurveFloatingPathEffect
import com.ddu.ui.effect.CurvePathFloatingAnimator
import com.ddu.ui.effect.ScaleFloatingAnimator
import com.ddu.ui.view.FloatingText
import com.iannotation.Element
import kotlinx.android.synthetic.main.fragment_ui_textview.*

/**
 * Created by yzbzz on 16/4/14.
 */
@Element("UI")
class TextViewFragment : DefaultFragment(), View.OnClickListener {

    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_ui_textview
    }

    override fun initView() {
        tv_reveal_text_view.setOnClickListener(this)
        tv_translateView.setOnClickListener(this)
        tv_curveView.setOnClickListener(this)
        tv_scaleView.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.tv_reveal_text_view -> tv_reveal_text_view.replayAnimation()
            R.id.tv_translateView -> showFloatingTextByTranslate(view)
            R.id.tv_curveView -> showFloatingTextByCurve(view)
            R.id.tv_scaleView -> showFloatingTextByScale(view)
            else -> {
            }
        }
    }

    private fun showFloatingTextByTranslate(view: View) {
        val translateFloatingText = FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.RED)
                .textSize(100)
                .textContent("+1000")
                .build()
        translateFloatingText.attach2Window()
        translateFloatingText.startFloating(view)
    }

    private fun showFloatingTextByCurve(view: View) {
        val cubicFloatingText = FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.RED)
                .textSize(100)
                .floatingAnimatorEffect(CurvePathFloatingAnimator())
                .floatingPathEffect(CurveFloatingPathEffect())
                .textContent("Hello! ").build()
        cubicFloatingText.attach2Window()
        cubicFloatingText.startFloating(view)
    }

    private fun showFloatingTextByScale(view: View) {
        val scaleFloatingText = FloatingText.FloatingTextBuilder(baseActivity)
                .textColor(Color.parseColor("#7ED321"))
                .textSize(100)
                .offsetY(-100)
                .floatingAnimatorEffect(ScaleFloatingAnimator())
                .textContent("+188").build()
        scaleFloatingText.attach2Window()
        scaleFloatingText.startFloating(view)
    }

    companion object {

        fun newInstance(taskId: String): TextViewFragment {
            val arguments = Bundle()
            arguments.putString(DefaultFragment.ARGUMENT_TASK_ID, taskId)
            val fragment = TextViewFragment()
            fragment.arguments = arguments
            return fragment
        }
    }
}
